src/zkbob/periphery/ZkBobPay.sol

```diff
@@ -89,13 +98,17 @@ contract ZkBobPay {
                 revert InsufficientAmount();
             }
         } else {
+            if (!enabledRouter[_router]) {
+                revert Unauthorized();
+            }
+
             uint256 balance = IERC20(token).balanceOf(address(this));
 
             if (_inToken != address(0)) {
-                IERC20(_inToken).approve(oneInchRouter, _inAmount);
+                IERC20(_inToken).approve(_router, _inAmount);
             }
 
-            (bool status,) = oneInchRouter.call{value: msg.value}(_oneInchData);
+            (bool status,) = _router.call{value: msg.value}(_routerData);
```

Comment:
in order to increase security does it make sense to keep of list of allowed methods for the particular router?

Reply:
Sure, not sure if it would make much sense in case of 1inch, since it will require supporting almost all selectors. But for li.fi this sounds reasonable. It looks like we only need to support one selector for `swapTokensGeneric(...)` - https://louper.dev/diamond/0x1231DEB6f5749EF6cE6943a275A1D3E7486F4EaE?network=polygon

Implemented in 85f55a9